flume-ng-core/src/test/java/org/apache/flume/client/avro/TestReliableSpoolingFileEventReader.java

```diff
@@ -137,7 +173,92 @@ public void testIgnorePattern() throws IOException {
     Assert.assertEquals("Expected 0, not: " + trackerFiles, 0,
         trackerFiles.size());
   }
+  
+  @Test
+  public void testIncludeExcludePatternNoConflict() throws IOException {
+    
+    // Expected behavior mixing include/exclude conditions:
+    // - file0, file1, file3: not deleted as matching ignore pattern and not
+    // matching include pattern
+    // - file2: deleted as not matching ignore pattern and matching include
+    // pattern
+    // - emptylineFile: not deleted as not matching ignore pattern but not
+    // matching include pattern as well
+    ReliableEventReader reader = new ReliableSpoolingFileEventReader.Builder()
+        .spoolDirectory(WORK_DIR)
+        .ignorePattern("^file[013]$")
+        .includePattern("^file2$")
+        .deletePolicy(DeletePolicy.IMMEDIATE.toString())
+        .build();
+ 
+    List<File> before = listFiles(WORK_DIR);
+    Assert.assertEquals("Expected 5, not: " + before, 5, before.size());
+
+    List<Event> events;
+    do {
+      events = reader.readEvents(10);
+      reader.commit();
+    } while (!events.isEmpty());
 
+
+    List<File> after = listFiles(WORK_DIR);
+    Assert.assertEquals("Expected 4, not: " + after, 4, after.size());
+        Set<String> expectedLeftFiles = new HashSet<String>(
```

Comment:
Wrong indentation, it should be in line with the `Assert` above.
Note: we use 2 spaces for indentation.
You can verify whether your code conforms to Flume's code formatting guidelines by running checkstyle on it: `mvn clean verify -DskipTests -Drat.skip`, please run it, there are a couple of other warnings too.